express-notes-api/index.js

```diff
@@ -0,0 +1,94 @@
+const express = require('express');
+const app = express();
+const fs = require('fs');
+
+const notes = require('./data.json');
+
+app.use(express.json());
+
+app.get('/api/notes', (req, res) => {
+  const resArr = [];
+  for (const prop in notes.notes) {
+    const resJSON = notes.notes[prop];
+    resArr.push(resJSON);
+  }
+  res.status(200).json(resArr);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  if (Number.isInteger(req.params.id)) {
+    if (req.params.id in notes.notes) {
+      res.status(200).json(notes.notes[req.params.id]);
+    } else {
+      res.status(404).json({ error: `cannot find note with id ${req.params.id}` });
+    }
+  } else {
+    res.status(400).json({ error: 'id must be a positive integer' });
+  }
+});
```

Comment:
What happens if `id` is negative? Have you tested these since you updated them?

Reply:
![image](https://user-images.githubusercontent.com/75236087/130312435-d601824b-8126-4ffc-877d-8340b3e26995.png)
![image](https://user-images.githubusercontent.com/75236087/130312455-d0022832-4310-4134-b0de-f640ae2477f7.png)

Updated to test.